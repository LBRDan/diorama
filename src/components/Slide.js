import React from 'react';
import PropTypes from 'prop-types';

import styles from './Slide.css';

const Slide = ({ children, style }) => (
  <div style={style} className={`${styles.slide} diorama-slide`}>
    {children}
  </div>
);

Slide.propTypes = {
  children: PropTypes.node.isRequired,
  style: PropTypes.shape({}),
};

Slide.defaultProps = {
  style: {},
};

export default Slide;
