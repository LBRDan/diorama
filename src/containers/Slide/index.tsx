/* eslint-disable react/no-unused-prop-types */
import React from "react";
import PropTypes from "prop-types";

import styles from "./Slide.module.css";

export type SlideProps = React.PropsWithChildren<{
	className?: string;
	style?: React.CSSProperties;
	notes?: string;
	// eslint-disable-next-line react/require-default-props
	index?: number;
	// eslint-disable-next-line react/require-default-props
	navigate?: (slideIndex: number) => void;
}>;

function Slide({ children, style, className }: SlideProps) {
	return (
		<div style={style} className={`${styles.slide} diorama-slide ${className}`}>
			<div className={`diorama-content ${styles.content}`}>{children}</div>
		</div>
	);
}

Slide.propTypes = {
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
	notes: PropTypes.string,
	style: PropTypes.shape({}),
};

Slide.defaultProps = {
	className: "",
	notes: undefined,
	style: {},
};

export default Slide;
