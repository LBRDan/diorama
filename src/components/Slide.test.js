import React from 'react';
import renderer from 'react-test-renderer';
import Slide from './Slide';

it('renders correctly', () => {
  const tree = renderer.create(<Slide>Slide content</Slide>).toJSON();
  expect(tree).toMatchSnapshot();
});
