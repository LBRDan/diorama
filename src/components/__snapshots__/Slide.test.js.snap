// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="slide diorama-slide"
  style={Object {}}
>
  Slide content
</div>
`;
